packages/driver/src/cy/commands/xhr.coffee

```diff
@@ -253,6 +253,9 @@ module.exports = (Commands, Cypress, cy, state, config) ->
     ## reset the existing server
     reset()
 
+    ## reset any state on the backend
+    Cypress.backend('reset:xhr:server')
```

Comment:
You can't do this because this function is async - Cypress will not await it and everything will be out of band causing race conditions.

You need to explicitly set up a handler for `test:before:run:async` and return this invocation. Refer to my previous messages and links to the other files that do the same things (like cookies).

Reply:
I thought of that, I figured this would actually work fine, since the "clear" command would be enqueued behind any "set" commands anyways. But I'll change it to be explicit instead of implicit, like you showed.